### PR TITLE
feat(board): associate WI with iteration using drag and drop in board…

### PR DESCRIPTION
**Pull request by vikram-raj**
… view.

Story: https://openshift.io/openshiftio/openshiftio/plan/detail/3c37c431-b0db-4c2f-941c-94e2bc2c18e0

This PR is only for board view.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by nimishamukherjee** [comment]
@vikram-raj do the iteration count get updated?

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by vikram-raj** [comment]
@nimishamukherjee Yes, it gets updated.

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

File: src/app/components/card/card.component.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ export class CardValue {
   avatar?: string;
   link?: string;
   menuItem?: any;
+  selfLink?: string;
```
Comment (sanbornsen) [inline src/app/components/card/card.component.ts]: These three are very work item specific. Take one object called **extraData** or something like that and have these three inside it.

Comment (vikram-raj) [inline src/app/components/card/card.component.ts]: Make sense. Will do this changes.

Comment (vikram-raj) [inline src/app/components/card/card.component.ts]: @sanbornsen This has been resolved here https://github.com/fabric8-ui/fabric8-planner/pull/2068/commits/0ea8061c6b989ed21feb04c520513379973b0d76

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2068-10 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2068-11 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2068-12 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Comment by nimishamukherjee** [comment]
@vikram-raj I'm using the snapshot url and cleared the cache
- The work item count is not being incremented
- The work item remains in the current iteration even after I have associated it with another iteration.

**Comment by vikram-raj** [comment]
@nimishamukherjee Work item gets associated with iteration but when I filter the work item with iteration it the API return wrong work item which is not associated with that iteration.

screenshot from 2017-07-13 12-30-04

**Comment by nimishamukherjee** [comment]
@pranavgore09 can you have a look at this. Can this be fixed before our demo?

**Comment by pranavgore09** [comment]
@nimishamukherjee @vikram-raj 
I am putting up the scenario regarding nested iterations and WI in it.

Consider there are two iterations `i1` and `i2`. And `i1` is the parent of `i2`.
Consider there are two work items `w1` and `w2`.

Now let's say we have assigned `i1` to `w1` and `i2` to `w2`.

When we filter on parent iteration `i1` -> we should get `w1` and `w2`. (Because work items from child iterations are also added in the filter)
When we filter on child iteration `i2` -> we should get only `w2`.

HTH.

**Comment by vikram-raj** [comment]
Thanks @pranavgore09

File: src/app/components/card/card.component.ts [hunk 2]
```diff
@@ -12,7 +12,8 @@ export class CardValue {
   hasLink: Boolean;
   avatar?: string;
   link?: string;
-  menuItem?: any;
+  menuItem?: object;
```
Comment (vikram-raj) [inline src/app/components/card/card.component.ts]: This has been resolved here https://github.com/fabric8-ui/fabric8-planner/pull/2068/commits/ebdc10199168c4454e1342df77a706770a442aef

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2068-29 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2068-fabric8-planner.badger.fabric8.io

**Pull request merged**